### PR TITLE
Milestone 3: Set up associations

### PR DESCRIPTION
**Pull request by Chrissiku**
## Project requirements
- ✔️  Create a class `Classroom` with the following:
  - [x] `@label` instance variable, should be initialized in the constructor.
  - [x] Setter and getter for `@label` (remember about `attr_accessor`).
- ✔️ Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - [x] Create the `has-many` side (a classroom has many students).
  - [x] Create the `belongs-to` side (a student belongs to a classroom).
  - [x] Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - [x] Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- ✔️ Create a class `Book` with the following:
  - [x] `@title` and `@author` instance variables, should be initialized in the constructor.
  - [x] Setters and getters for instance variables (remember about `attr_accessor`).
- ✔️ Create a class `Rental` with the following:
  - [x] `@date` instance variable, should be initialized in the constructor.
  - [x] Setter and getter for `@date` (remember about `attr_accessor`).
- ✔️ Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - [x] Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - [x] Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - [x] Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - [x] Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - [x] Modify the constructor of `Rental` so `Book` and `Person` are set in it.

## Show your support
Give a ⭐️ if you like this project!

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_reader :rental
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
```
Comment (thecodechaser) [inline book.rb]: - [x] Please initialize a `@rental = [ ]` variable on the constructor where you save your rentals on line 12.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+    bool.rental << self
```
Comment (thecodechaser) [inline rental.rb]: - [x] There's a typo, Please change `bool` to `book`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Chrissiku👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Chrissiku) [reply book.rb]: Of course, 
Thank you @thecodechaser

File: rental.rb (hunk 2 above)
Comment (Chrissiku) [reply rental.rb]: Oh wow, Thank you a lot

**Review by dasileker — APPROVED** [review]
### **Hi @Chrissiku,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chrissiku** [comment]
Thank you @dasileker

**Pull request merged**